inst/logistic.stan

```diff
@@ -1,7 +1,7 @@
 data {
   int<lower=0> N;
   int<lower=0> K;
-  int<lower=0,upper=1> y[N];
+  array[N] int<lower=0, upper=1> y;
```

Comment:
2.29 will warn when using the old array syntax and in order to avoid having warnings in unit tests, I fixed these test models.